src/APIController.react.js

```diff
@@ -70,7 +73,19 @@ class UnconnectedContainer extends Component {
             // Hasn't already hydrated
             appLifecycle === getAppState('STARTED')
         ) {
-            dispatch(hydrateInitialOutputs());
+            let errorLoading = false;
+            try {
+                dispatch(hydrateInitialOutputs());
+            } catch (err) {
+                errorLoading = true;
+            } finally {
+                this.setState(
+                    state =>
+                        state.errorLoading !== errorLoading
+                            ? {errorLoading}
+                            : null
+                );
+            }
```

Comment:
React 16 component render differs greatly from React 15 in terms of error handling. `MultiGraph.overallOrder()` called by `hydrateInitialOutputs` throws an error when there is a cycle. This error needs to be caught and handled, otherwise the entire app fails to render.